src/common/meta/dashboards/reports.rs

```diff
@@ -199,7 +199,7 @@ impl Default for Report {
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct ReportEmailDetails {
-    pub recepients: Vec<String>,
+    pub recipients: Vec<String>,
```

Comment:
Thank you, most of the change are fine, only this piece we can't simple rename it, it will cause compatible issue. because already some data stored with this name.

Reply:
I can add a serde alias to support deserialising the old name?

Reply:
We can, but with that we need more compatible testing, can't simple merge it.

Reply:
I added to my comment "And add tests to ensure this works and doesnt regress."

Is that sufficient?

Or, what else is needed?

Reply:
If you want to rename it, no problem, Please help run some testing and confirm it will work.
first, install o2 and create and report with email destination.
then, switch to this branch and check if the data still working.

Reply:
@hengfeiyang , I created a unit test to ensure it is not accidentally broken.

Also added .typos.yaml which is used by https://crates.io/crates/typos-cli , and included some more typos, excluding the frontend typos to prevent this PR getting too complex.

I will follow up with another PR for the frontend code, after this one is merged.